### PR TITLE
Update Jasmine, Sinon & Karma test runner

### PR DESCRIPTION
**Pull request by alexgibson**

### CODE REVIEW

File: tests/unit/spec/utils.js [hunk 1]
```diff
@@ -1,10 +1,13 @@
...
 
-describe("utils.js", function() {
+/* global describe, beforeEach, afterEach, it, expect, sinon, spyOn */
+
+describe('utils.js', function() {
+
+    'use strict';
 
     var macUA = 'Mozilla/5.0 (Macintosh; Intel Mac OS X 10.10; rv:36.0) Gecko/20100101 Firefox/36.0';
     var winUA = 'Mozilla/5.0 (Windows NT 6.1; rv:36.0) Gecko/20100101 Firefox/36.0';
```
Comment (schalkneethling) [inline tests/unit/spec/utils.js]: I wonder, can we use require and perhaps load in lib/config.js where we also have UA defined among other things? Perhaps even create a new file in lib, with common strings etc. used across the unit and functional tests?

Might be beyond the scope of this PR, but though I'd throw it out there.

Comment (alexgibson) [inline tests/unit/spec/utils.js]: A good idea I think, makes sense to me :+1: 

Might be worthy of a follow up PR, but I won't let it block this one :)

**Comment by schalkneethling** [comment]
All the tests passes, and dang quickly as well \o/

One thing though, we are getting some warning about 404's printed to the console:

WARN [web-server]: 404: /browser-windows.png
WARN [web-server]: 404: /browser-mac.png
................WARN [web-server]: 404: /browser-linux.png
WARN [web-server]: 404: /browser-ios.png
WARN [web-server]: 404: /browser-android.png
WARN [web-server]: 404: /high-res.png
WARN [web-server]: 404: /browser-mac-high-res.png
WARN [web-server]: 404: /low-res.png
WARN [web-server]: 404: /foo.png

**Comment by schalkneethling** [comment]
Other than the above, and the one question regarding sharing some common strings across the unit and functional tests, this is a r+ :moneybag:

**Comment by alexgibson** [comment]
> One thing though, we are getting some warning about 404's printed to the console:

WARN [web-server]: 404: /browser-windows.png
WARN [web-server]: 404: /browser-mac.png
................WARN [web-server]: 404: /browser-linux.png
WARN [web-server]: 404: /browser-ios.png
WARN [web-server]: 404: /browser-android.png
WARN [web-server]: 404: /high-res.png
WARN [web-server]: 404: /browser-mac-high-res.png
WARN [web-server]: 404: /low-res.png
WARN [web-server]: 404: /foo.png

Yep, those are some non existent images that get used in a couple of the tests - would like a way to suppress them, but they're pretty harmless.

File: tests/unit/spec/utils.js (hunk 1 above)
Comment (schalkneethling) [inline tests/unit/spec/utils.js]: Agreed

**Comment by schalkneethling** [comment]
:tropical_fish:

**Pull request merged**